main.nf

```diff
@@ -29,23 +29,30 @@ if (! bamfile.exists()) {
 
 
 if ( params.run_manta ) {
-    process index_bamfile {
-        input:
-            file 'bamfile' from bamfile
-        output:
-            file 'bamfile.bai' into bamfile_index
+    bamindex = infer_bam_index_from_bam()
+    if (!bamindex) {
+        process index_bamfile {
+            input:
+                file 'bamfile' from bamfile
+            output:
+                file 'bamfile.bai' into bamfile_index
 
-        module 'bioinfo-tools'
-        module "$params.modules.samtools"
+            module 'bioinfo-tools'
+            module "$params.modules.samtools"
 
-        // We only need one core for this part
-        clusterOptions = {
-            "-A $params.project -p core"
-        }
+            publishDir 'results'
```

Comment:
Is this something that is needed because the bam file is used or something? Does that need documenting, or is it self-explanatory?